websocket-sharp/DestroyOnDisposeWebSocket.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using WebSocketSharp.Net.WebSockets;
+
+namespace WebSocketSharp
+{
+    public class DestroyOnDisposeWebSocket: WebSocket, IDisposable
+    {
+        internal DestroyOnDisposeWebSocket(HttpListenerWebSocketContext context, string protocol) : base(context, protocol)
+        {
+        }
+
+        internal DestroyOnDisposeWebSocket(TcpListenerWebSocketContext context, string protocol) : base(context, protocol)
+        {
+        }
+
+        public DestroyOnDisposeWebSocket(string url, params string[] protocols) : base(url, protocols)
+        {
+        }
+        
+        #region Explicit Interface Implementations
+
+        /// <summary>
+        /// Closes the WebSocket connection, and releases all associated resources, without attempting
+        /// the disconnection handshake
+        /// </summary>
+        /// <remarks>
+        /// This method closes the connection with <see cref="CloseStatusCode.Away"/>.
+        /// </remarks>
+        void IDisposable.Dispose ()
+        {
+            close (new CloseEventArgs (CloseStatusCode.Away), false, true);
```

Comment:
why not set the wait flag to false too here? Is there any benefit in potentially blocking the thread (granted it is not on the main thread) when we might not care about the result

Reply:
Nope, there isn't. The bool there doesn't take any effect

        send = send && _readyState == WebSocketState.Open;
        wait = wait && send;

So i just ignored it. I could change it to false.